src/main/java/task/DeadLine.java

```diff
@@ -0,0 +1,81 @@
+package task;
+
+import dukeexception.DukeException;
+import utils.DateTimeUtils;
+
+/**
+ *  Handles all deadLines.
+ */
+public class DeadLine extends Task {
+    /**
+     * Store for deadLine date.
+     */
+    private String dateTime;
+
+    /**
+     * Private constructor.
+     *
+     * @param str Task name.
+     * @param checked Boolean for whether current task is marked.
+     * @param dateTime DeadLine date.
+     */
+    private DeadLine(String str, boolean checked, String dateTime) {
+        super(str, checked);
+        this.dateTime = dateTime;
+    }
+
+    /**
+     * Factory method that returns deadLine task from user input.
+     *
+     * @param inputLine Input from user.
+     * @return DeadLine Task.
+     * @throws DukeException Checks valid input from user.
+     */
+    public static DeadLine generate(String[] inputLine) throws DukeException {
+        if (inputLine.length < 2) {
+            throw new DukeException("Missing description");
+        }
+        String[] toPrintSplit = inputLine[1].split(" /by ", 2);
+        if (toPrintSplit.length < 2) {
+            throw new DukeException("Missing deadline");
+        }
+        return new DeadLine(toPrintSplit[0], false, DateTimeUtils.dateFormatter(toPrintSplit[1]));
```

Comment:
I like your clever usage of splitting the input line to check the errors, it is easy to read and understands.

Reply:
Thanks!